python/az/aro/azext_aro/_aad.py

```diff
@@ -54,6 +59,15 @@ def get_service_principal(self, app_id):
         return None
 
     def create_service_principal(self, app_id):
-        return self.client.service_principals.create(ServicePrincipalCreateParameters(
-            app_id=app_id,
-        ))
+        max_retries = 3
+        retries = 0
+        while True:
+            try:
+                return self.client.service_principals.create(
+                    ServicePrincipalCreateParameters(app_id=app_id))
+            except GraphErrorException:
+                if retries >= max_retries:
+                    raise
+                retries += 1
+                logger.warning("%s; retry %d of %d", GraphErrorException, retries, max_retries)
```

Comment:
Might be better to log the exception instance rather than the exception class?
```suggestion
            except GraphErrorException as ex:
                if retries >= max_retries:
                    raise
                retries += 1
                logger.warning("%s; retry %d of %d", ex, retries, max_retries)
```

Reply:
whoops

Reply:
Fixed, thanks